main.ts

```diff
@@ -8,7 +8,8 @@ export default class NeighbouringFileNavigator extends Plugin {
 
 	navigateToNeighbouringFile(next?: boolean) {
 		const activeFile = this.app.workspace.getActiveFile();
-		const files = activeFile?.parent?.children;
+		const files = activeFile?.parent?.children?.filter(file => 'basename' in file);
```

Comment:
A quick test shows that files with other extensions are also opened.

The primary purpose of this plugin is to efficiently navigate markdown files, opening files with external programs would interrupt this behaviour. Therefore I'd change the filter to something more restrictive.
```suggestion
		const files = activeFile?.parent?.children?.filter(file => file.extension === 'md');
```